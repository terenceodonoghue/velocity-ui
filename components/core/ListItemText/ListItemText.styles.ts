import { css } from '@emotion/react';

const listItemText = css`
  margin-left: 16px;
`;

export default listItemText;
