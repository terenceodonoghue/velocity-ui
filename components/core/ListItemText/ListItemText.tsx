import React, { FunctionComponent } from 'react';
import listItemText from './ListItemText.styles';
import type { ListItemTextProps } from './ListItemText.types';

const ListItemText: FunctionComponent<ListItemTextProps> = ({
  children,
  ...props
}) => (
  <span css={listItemText} data-testid="listitemtext" {...props}>
    {children}
  </span>
);

export default ListItemText;
