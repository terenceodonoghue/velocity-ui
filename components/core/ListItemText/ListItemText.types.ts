import { HTMLAttributes } from 'react';

export interface ListItemTextProps extends HTMLAttributes<HTMLSpanElement> {
  selected?: boolean;
}
