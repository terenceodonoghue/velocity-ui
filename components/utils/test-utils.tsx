import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
