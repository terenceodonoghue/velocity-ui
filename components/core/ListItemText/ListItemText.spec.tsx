import React from 'react';
import faker from 'faker';
import { render } from 'test-utils';
import ListItemText from './ListItemText';

describe('ListItemText', () => {
  it('renders content', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId } = render(
      <ListItemText>{content}</ListItemText>,
    );

    expect(getByTestId('listitemtext')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });
});
