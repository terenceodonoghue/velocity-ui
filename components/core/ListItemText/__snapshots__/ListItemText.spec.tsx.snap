// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemText renders content 1`] = `
.emotion-0 {
  margin-left: 16px;
}

<div>
  <span
    class="emotion-0"
    data-testid="listitemtext"
  >
    The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
  </span>
</div>
`;
